<launch>
    <arg name="marker_size" doc="Size of the ArUco marker used, in meters" value="0.1"/>
    <arg name="marker_id" doc="The ID of the ArUco marker used" value="10"/>

    <!-- start Zivid camera driver node -->
    <node name="zivid_camera" pkg="zivid_camera" type="zivid_camera_node" output="screen" ns="zivid_camera">
        <param name="frame_id" value="zivid_optical_frame" />
    </node>

    <!-- start ArUco -->
    <node name="aruco_tracker" pkg="aruco_ros" type="single">
        <remap from="/camera_info" to="/zivid_camera/color/camera_info"/>
        <remap from="/image" to="/zivid_camera/color/undistorted_image_color"/>
        <param name="image_is_rectified" value="true"/>
        <param name="marker_size"        value="$(arg marker_size)"/>
        <param name="marker_id"          value="$(arg marker_id)"/>
        <param name="reference_frame"    value="zivid_optical_frame"/>
        <param name="camera_frame"       value="zivid_optical_frame"/>
        <param name="marker_frame"       value="camera_marker"/>
    </node>

    <!-- start the capture node -->
    <node name="capture_2d" pkg="easy_handeye_calibration" type="sample_capture_2d.py">
    </node>
</launch>
